### PR TITLE
Fix todo

### PR DESCRIPTION
**Pull request by davgur**

### CODE REVIEW

File: api/handlers.go [hunk 1]
```diff
@@ -93,18 +104,44 @@ func (a *App) handleGetPlaylists(c *gin.Context) {
...
+		playlistDTOByID[playlistID].MaxItemPosition = maxPosition
 	}
 
 	if err := rows.Err(); err != nil {
 		errs.NewInternalError(pkgerr.WithStack(err)).Abort(c)
 		return
 	}
 
+	rows, err = models.NewQuery(
+		qm.Select(fmt.Sprintf("DISTINCT ON(%[1]s) %[1]s, %[2]s", models.PlaylistItemColumns.PlaylistID, models.PlaylistItemColumns.ContentUnitUID)),
```
Comment (davgur) [inline api/handlers.go:122]: I'm not sure about this (I need image for playlist for show on the client)

File: databases/mdb/consts.go [hunk 2]
```diff
@@ -1,76 +0,0 @@
-package mdb
```
Comment (davgur) [inline databases/mdb/consts.go:1]: I remove it becouse Use only on the tests

File: lib/chronicles/chronicles.go [hunk 3]
```diff
@@ -27,7 +27,7 @@ import (
 )
 
 const (
-	SCAN_SIZE     = 1000
+	SCAN_SIZE     = 100
```
Comment (davgur) [inline lib/chronicles/chronicles.go:30]: if it too big return with server time out

File: middleware/auth.go [hunk 4]
```diff
@@ -150,9 +151,8 @@ func parseToken(r *http.Request) string {
 	return ""
 }
 
-func getOrCreateUser(exec boil.Executor, claims *IDTokenClaims) (*models.User, error) {
-	//TODO: wrap in transaction
-	user, err := models.Users(models.UserWhere.AccountsID.EQ(claims.Sub)).One(exec)
+func getOrCreateUser(db *sql.DB, claims *IDTokenClaims) (*models.User, error) {
+	user, err := models.Users(models.UserWhere.AccountsID.EQ(claims.Sub)).One(db)
```
Comment (edoshor) [inline middleware/auth.go:155]: this should be inside the transaction

File: middleware/auth.go [hunk 5]
```diff
@@ -168,8 +168,16 @@ func getOrCreateUser(exec boil.Executor, claims *IDTokenClaims) (*models.User, e
...
+			return pkgerr.Wrap(err, "create new user in DB")
+		}
+		return nil
+	})
+
+	if err != nil {
+		return nil, err
 	}
-	return user, nil
+	err = user.Reload(db)
```
Comment (edoshor) [inline middleware/auth.go:181]: why we need this here ?

File: sqlboiler.toml [hunk 6]
```diff
@@ -7,6 +7,6 @@ blacklist=["schema_migrations"]
   dbname="mydb"
   host="localhost"
   port=5432
-  user="user"
-  pass="password"
+  user="postgres"
```
Comment (edoshor) [inline sqlboiler.toml:10]: why change that ?
this is also used in docker-compose.yml

**Review by edoshor — CHANGES_REQUESTED** [review]

File: api/handlers.go [hunk 7]
```diff
@@ -56,12 +56,19 @@ func (a *App) handleGetPlaylists(c *gin.Context) {
...
+	}
+
 	_, offset := appendListMods(&mods, r.ListRequest)
 	if int64(offset) >= total {
 		concludeRequest(c, NewPlaylistsResponse(0, 0), nil)
 		return
 	}
 
+	if r.ExistUnit != "" {
+		mods = append(mods, qm.Load("PlaylistItems", models.PlaylistItemWhere.ContentUnitUID.EQ(r.ExistUnit)))
```
[the remark this answers is not in the document]
Comment (davgur) [reply api/handlers.go:70]: it's filter playlists that have unit on playlist_items
It's need for mark playlists that already have this unit
image

File: api/handlers.go [hunk 8]
```diff
@@ -879,8 +965,9 @@ func makePlaylistDTO(playlist *models.Playlist) *Playlist {
 
 		resp.Items = make([]*PlaylistItem, resp.TotalItems)
 		sort.SliceStable(playlist.R.PlaylistItems, func(i int, j int) bool {
-			return playlist.R.PlaylistItems[i].Position < playlist.R.PlaylistItems[j].Position
+			return playlist.R.PlaylistItems[i].Position > playlist.R.PlaylistItems[j].Position
 		})
+		resp.MaxItemPosition = playlist.R.PlaylistItems[0].Position
```
[the remark this answers is not in the document]
Comment (davgur) [reply api/handlers.go:970]: so I will need fetch all playlist items of all playlists for every content unit

File: api/models.go [hunk 9]
```diff
@@ -138,6 +143,11 @@ type Reaction struct {
 	SubjectUID  string `json:"subject_uid"`
 }
 
+type ReactionCount struct {
+	Total string `json:"total"`
+	Reaction
```
[the remark this answers is not in the document]
Comment (davgur) [reply api/models.go:148]: fix

File: api/handlers.go [hunk 10]
```diff
@@ -496,16 +533,28 @@ func (a *App) handleGetReactions(c *gin.Context) {
...
 
 	db := c.MustGet("MY_DB").(*sql.DB)
 	total, err := models.Reactions(mods...).Count(db)
 	if err != nil {
 		errs.NewInternalError(pkgerr.WithStack(err)).Abort(c)
 		return
 	}
-
+	if r.OrderBy == "" {
+		r.OrderBy = fmt.Sprintf("%s DESC", models.ReactionColumns.ID)
```
[the remark this answers is not in the document]
Comment (davgur) [reply api/handlers.go:539]: was think that all time we need that latest reactions will be first
so client must to do it?

File: api/handlers.go [hunk 11]
```diff
@@ -584,35 +633,56 @@ func (a *App) handleRemoveReactions(c *gin.Context) {
 
 	db := c.MustGet("MY_DB").(*sql.DB)
 	err := sqlutil.InTx(context.TODO(), db, func(tx *sql.Tx) error {
-		_, err := models.Reactions(models.ReactionWhere.UserID.EQ(user.ID),
+		count, err := models.Reactions(models.ReactionWhere.UserID.EQ(user.ID),
```
[the remark this answers is not in the document]
Comment (davgur) [reply api/handlers.go:619]: fix

File: api/handlers.go [hunk 12]
```diff
@@ -584,35 +633,56 @@ func (a *App) handleRemoveReactions(c *gin.Context) {
...
+		}
+		return nil
 	})
 
 	concludeRequest(c, nil, err)
 }
 
-func (a *App) handleLikeCount(c *gin.Context) {
-	// TODO: count per kind
+func (a *App) handleReactionCount(c *gin.Context) {
```
[the remark this answers is not in the document]
Comment (davgur) [reply api/handlers.go:635]: units have few content types, so I will need bring list of units types? 
Ore create new content types (Collections, sources units ...)?

[the remark this answers is not in the document]
Comment (davgur) [reply api/handlers.go:635]: fixed

File: api/handlers.go [hunk 13]
```diff
@@ -584,35 +633,56 @@ func (a *App) handleRemoveReactions(c *gin.Context) {
...
 		return
 	}
+	defer rows.Close()
+
+	resp := make([]*ReactionCount, 0)
+	for rows.Next() {
+		r := &ReactionCount{}
+		err = rows.Scan(r.SubjectUID, r.SubjectType, r.Kind, r.Total)
+		resp = append(resp, r)
+	}
```
[the remark this answers is not in the document]
Comment (davgur) [reply api/handlers.go:666]: fix

File: middleware/auth.go (hunk 5 above)
Comment (davgur) [reply middleware/auth.go:181]: removed

File: api/handlers.go [hunk 14]
```diff
@@ -636,6 +706,9 @@ func (a *App) handleGetHistory(c *gin.Context) {
 		errs.NewInternalError(pkgerr.WithStack(err)).Abort(c)
 		return
 	}
+	if r.OrderBy == "" {
+		r.OrderBy = fmt.Sprintf("%s DESC", models.HistoryColumns.ChroniclesTimestamp)
```
[the remark this answers is not in the document]
Comment (davgur) [reply api/handlers.go:693]: add

File: api/handlers.go [hunk 15]
```diff
@@ -682,14 +755,20 @@ func (a *App) handleDeleteHistory(c *gin.Context) {
 
 	db := c.MustGet("MY_DB").(*sql.DB)
 	err = sqlutil.InTx(context.TODO(), db, func(tx *sql.Tx) error {
-		_, err := models.Histories(models.HistoryWhere.UserID.EQ(user.ID), models.HistoryWhere.ID.EQ(id)).DeleteAll(tx)
-		return err
+		count, err := models.Histories(models.HistoryWhere.UserID.EQ(user.ID), models.HistoryWhere.ID.EQ(id)).DeleteAll(tx)
```
[the remark this answers is not in the document]
Comment (davgur) [reply api/handlers.go:741]: fix

File: middleware/auth.go (hunk 4 above)
Comment (davgur) [reply middleware/auth.go:155]: fix
why becouse of atomic read and edit? 
but it's on the single tread all time

File: sqlboiler.toml (hunk 6 above)
Comment (davgur) [reply sqlboiler.toml:10]: I was not unestand why its on the git, 
add to it .gitignore
push changes by mistake

File: middleware/auth.go (hunk 4 above)
Comment (davgur) [reply middleware/auth.go]: hm...
I dont find othe solution
Maybe only add special client call that will create user

File: api/handlers.go [hunk 16]
```diff
@@ -56,12 +56,19 @@ func (a *App) handleGetPlaylists(c *gin.Context) {
 		return
 	}
 
+	if r.ListRequest.PageSize == 0 {
```
[the remark this answers is not in the document]
Comment (davgur) [reply api/handlers.go]: removed

File: .gitignore [hunk 17]
```diff
@@ -13,3 +13,4 @@
 _vendor-*/
 
 archive-my
+sqlboiler.toml
```
Comment (davgur) [inline .gitignore:16]: need to remove this?

File: api/models.go [hunk 18]
```diff
@@ -107,23 +110,25 @@ type SubscriptionsResponse struct {
 }
 
 type SubscribeRequest struct {
-	CollectionUID  string `json:"collection_uid" binding:"omitempty"`
-	ContentType    string `json:"content_type" binding:"omitempty"`
-	ContentUnitUID string `json:"content_unit_uid" binding:"omitempty"`
+	CollectionUID  string `json:"collection_uid" form:"collection_uid" binding:"omitempty"`
```
[the remark this answers is not in the document]
Comment (davgur) [reply api/models.go:118]: blya, look like I undestend what you say
I have few same places

[the remark this answers is not in the document]
Comment (davgur) [reply api/models.go:118]: hm, or not undestand

[the remark this answers is not in the document]
Comment (davgur) [reply api/models.go:118]: I was all time think that it's FROM and not FORM

File: api/handlers.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (edoshor) [reply api/handlers.go:66]: Now that I understand what this is for please add a comment to clarify why we are doing this here

**Pull request merged**